core/src/main/java/org/elasticsearch/action/support/broadcast/node/TransportBroadcastByNodeAction.java

```diff
@@ -134,7 +134,7 @@ private final Response newResponse(Request request, AtomicReferenceArray respons
         }
         totalShards += unavailableShardExceptions.size();
         int failedShards = exceptions.size();
-        return newResponse(request, totalShards, successfulShards, failedShards, broadcastByNodeResponses, exceptions);
+        return newResponse(request, totalShards, successfulShards, failedShards, broadcastByNodeResponses, exceptions, clusterService.state());
```

Comment:
can we tunnel through the cluster state used in the action? that way we don't need to worry about consistency between the two (concurrent index deletion etc.)